iphone/Maps/Classes/ClipboardMonitor.mm

```diff
@@ -0,0 +1,107 @@
+#import "MapsAppDelegate.h"
+#import "ClipboardMonitor.h"
+#import "UIKitCategories.h"
+#import "MapsAppDelegate.h"
+#import <CoreApi/DeepLinkParser.h>
+#include <CoreApi/Framework.h>
+#import <CoreApi/DeepLinkData.h>
+#import "UIKitCategories.h"
+#import <CoreApi/DeepLinkSearchData.h>
+
+@interface ClipboardWatcher ()
+
+@property (nonatomic, strong) NSTimer *timer;
+@property (nonatomic, copy) NSString *previousClipboardText;
+@property (nonatomic, assign) BOOL isErrorPromptShown;
+
+@end
+
+@implementation ClipboardWatcher
```

Comment:
Why watcher? Isn't it enough to check the pasteboard once on the app foreground activation?

Reply:
That's only the name of the implementation, it runs only once whenever someone enters foreground from background (or changes app and even a cold start).